node-sdk: asset-mgmt-with-roles.js: panic: runtime error: invalid memory address or nil pointer dereference

## Description

Chaincode container crashes when running UT asset-mgmt-with-roles.js in network mode

I have a 4-peer network setup with security and pbft batch consensus:
152dc2ebc842 peer 9.42.91.158:30085->5000/tcp,_9.42.91.158:30084->30303/tcp 
f98863fdad33 peer  9.42.91.246:30143->5000/tcp,_9.42.91.246:30142->30303/tcp  
28723cea2bbf peer  9.42.91.158:30083->5000/tcp,_9.42.91.158:30082->30303/tcp
c026a7d9bfbc peer 9.42.91.246:30141->5000/tcp,_9.42.91.246:30140->30303/tcp
b6ed3632c0cf ca     9.42.91.246:30139->30303/tcp,_9.42.91.246:30138->50051/tcp

During the AssetManagementChaincode] Init, I hit this:
panic: runtime error: invalid memory address or nil pointer dereference
    panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x48a87c]

goroutine 23 [running]:
panic(0x9856c0, 0xc820012080)
    /opt/go/src/runtime/panic.go:464 +0x3e6
build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim.(_Handler).serialSend(0xc8201d0380, 0x0, 0x0, 0x0)
    /opt/gopath/src/build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim/handler.go:73 +0x52c
build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim.(_Handler).handleQuery.func1.1(0xc8201d0380, 0xc820047e20)
    /opt/gopath/src/build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim/handler.go:319 +0x2e
panic(0x9856c0, 0xc820012080)
    /opt/go/src/runtime/panic.go:426 +0x4e9
main.(_AssetManagementChaincode).Query(0xd647d8, 0xc820181160, 0xc8202024b5, 0x5, 0xc8202024d0, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
    /opt/gopath/src/build-chaincode/asset_management_with_roles.go:263 +0x615
build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim.(_Handler).handleQuery.func1(0xc8201d0380, 0xc8201292c0)
    /opt/gopath/src/build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim/handler.go:340 +0x5d4
created by build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim.(*Handler).handleQuery
    /opt/gopath/src/build-chaincode/vendor/github.com/hyperledger/fabric/core/chaincode/shim/handler.go:356 +0x3f
## Describe How to Reproduce

  Start 4 validating peers in network node, running pbft batch.
  Run the UT asset_management_with_roles, using the ca server and peer0 grpc addresses.
[dev-vp0-7a9d5740.txt](https://github.com/hyperledger/fabric/files/328621/dev-vp0-7a9d5740.txt)
[dev-vp1-7a9d5740.txt](https://github.com/hyperledger/fabric/files/328620/dev-vp1-7a9d5740.txt)
[dev-vp2-7a9d5740.txt](https://github.com/hyperledger/fabric/files/328623/dev-vp2-7a9d5740.txt)
[dev-vp3-7a9d5740.txt](https://github.com/hyperledger/fabric/files/328625/dev-vp3-7a9d5740.txt)
[hlc.debug.txt](https://github.com/hyperledger/fabric/files/328624/hlc.debug.txt)
[vp0.txt](https://github.com/hyperledger/fabric/files/328622/vp0.txt)
[vp1.txt](https://github.com/hyperledger/fabric/files/328627/vp1.txt)
[vp2.txt](https://github.com/hyperledger/fabric/files/328628/vp2.txt)
[vp3.txt](https://github.com/hyperledger/fabric/files/328626/vp3.txt)

--- a/examples/chaincode/go/asset_management_with_roles/asset_management_with_roles.go
+++ b/examples/chaincode/go/asset_management_with_roles/asset_management_with_roles.go
@@ -260,7 +260,7 @@ func (t *AssetManagementChaincode) Query(stub *shim.ChaincodeStub, function stri
 	}
 
 	if len(row.Columns) == 0 {
-		jsonResp := "{\"Error\":\"Failed retrieving owner for " + asset + ". Error " + err.Error() + ". \"}"
+		jsonResp := "{\"Error\":\"Failed retrieving owner for " + asset + ". \"}"
 		return nil, errors.New(jsonResp)
 	}
 

--- a/examples/chaincode/go/asset_management_with_roles/asset_management_with_roles_test.go
+++ b/examples/chaincode/go/asset_management_with_roles/asset_management_with_roles_test.go
@@ -167,6 +167,12 @@ func TestAssetManagement(t *testing.T) {
 	if !reflect.DeepEqual(theOnwerIs, bobAccount) {
 		t.Fatal("Bob is not the owner of Picasso")
 	}
+
+	// Check who is the owner of an asset that doesn't exist
+	_, err = whoIsTheOwner("Klee")
+	if err == nil {
+		t.Fatal("This asset doesn't exist. Querying should fail.")
+	}
 }
 
 func deploy(admCert crypto.CertificateHandler) error {

